### DISCUSSION TITLE
Incorrect/Bypass Custom filtering rules???

### DISCUSSION THREAD

**Discussion by t0ny54** [AdguardTeam/AdGuardHome#4200] [Q&A]
Server IP, DNS and bind hosts AGH : x.x.x.x
(I don't use 0.0.0.0 for bind host AGH)

Custom filtering rules => block example.org
example.org are block in Client Side, but in Server Side example.org are bypass/not blocked.
Query log server side example.org missing and query log client not missing.

Windows OS
AGH 0.107.3

Thank you

**Comment by ainar-g** [comment]
Please add the following information:

 *  What is the exact rule you've added to the Custom filtering rules?

 *  What do you mean when you say “client side” and “server side”?  Do you mean that the request from a client gets filtered, but a request from the server itself isn't?

**Reply by t0ny54** [reply]
- What is the exact rule you've added to the Custom filtering rules?

||example.org^$important

- What do you mean when you say “client side” and “server side”? Do you mean that the request from a client gets filtered, but a request from the server itself isn't?

Yes...

DNS blocklists filtering in client and server don't have issue, but Custom filtering rules are problem...

**Reply by ainar-g** [reply]
Did you set AdGuard Home up as the server's DNS resolver?  Because the OS of the server could be thought of as another client.  You can check that by running:

```sh
nslookup -debug -type=a example.com
```

And then:

```sh
nslookup -debug -type=a example.com "$YOUR_AGH_BIND_IP_ADDR"
```

**Reply by t0ny54** [reply]
> ```shell
> nslookup -debug -type=a example.com "$YOUR_AGH_BIND_IP_ADDR"
> ```

No, i don't set AGH to my primery dns resolver...

C:\Users\link>nslookup -debug -type=a pasino.com
------------
Got answer:
    HEADER:
        opcode = QUERY, id = 1, rcode = NXDOMAIN
        header flags:  response, recursion avail.
        questions = 1,  answers = 0,  authority records = 0,  additional = 1

    QUESTIONS:
        1.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.ip6.arpa, type = PTR, class = IN
    ADDITIONAL RECORDS:
    ->  (root)
        ??? unknown type 41 ???
        ttl = 0 (0 secs)

------------
Server:  UnKnown
Address:  ::1

------------
Got answer:
    HEADER:
        opcode = QUERY, id = 2, rcode = NOERROR
        header flags:  response, want recursion, recursion avail.
        questions = 1,  answers = 2,  authority records = 0,  additional = 1

    QUESTIONS:
        pasino.com, type = A, class = IN
    ANSWERS:
    ->  pasino.com
        internet address = 172.67.130.203
        ttl = 2400 (40 mins)
    ->  pasino.com
        internet address = 104.21.9.42
        ttl = 2400 (40 mins)
    ADDITIONAL RECORDS:
    ->  (root)
        ??? unknown type 41 ???
        ttl = 0 (0 secs)

------------
Non-authoritative answer:
Name:    pasino.com
Addresses:  172.67.130.203
          104.21.9.42


C:\Users\link>nslookup -debug -type=a pasino.com "192.168.1.2"
------------
Got answer:
    HEADER:
        opcode = QUERY, id = 1, rcode = NOERROR
        header flags:  response, want recursion, recursion avail.
        questions = 1,  answers = 1,  authority records = 0,  additional = 0

    QUESTIONS:
        2.1.168.192.in-addr.arpa, type = PTR, class = IN
    ANSWERS:
    ->  2.1.168.192.in-addr.arpa
        name = MiniPC
        ttl = 60 (1 min)

------------
Server:  MiniPC
Address:  192.168.1.2

------------
Got answer:
    HEADER:
        opcode = QUERY, id = 2, rcode = NOERROR
        header flags:  response, want recursion, recursion avail.
        questions = 1,  answers = 1,  authority records = 0,  additional = 0

    QUESTIONS:
        pasino.com, type = A, class = IN
    ANSWERS:
    ->  pasino.com
        internet address = 192.168.1.2
        ttl = 60 (1 min)

------------
Non-authoritative answer:
Name:    pasino.com
Address:  192.168.1.2

**Reply by ainar-g** [reply]
>  No, i don't set ADH to my primery dns resolver...

AGH cannot filter the traffic that goes around it.  The machine on which AGH is installed should also use AGH as its default DNS resolver.

**Reply by t0ny54** [reply]
> > No, i don't set ADH to my primery dns resolver...
> 
> AGH cannot filter the traffic that goes around it. The machine on which AGH is installed should also use AGH as its default DNS resolver.

But DNS filtering work well on my side. Only custom filering not work well.
Oh never mind...

Thank you for your answer...
